falcon/api.py

```diff
@@ -418,14 +418,17 @@ def my_serializer(req, exception):
                 preferred = req.client_prefers(('application/x-yaml',
                                                 'application/json'))
 
+                headers = []
+
                 if preferred is not None:
+                    headers.append(('append', 'Vary', 'Accept'))
```

Comment:
Having to specify 'append' seems a bit clunky to me and would require additional documentation to explain. I'm starting to wonder if it would be more straightforward, after all, to just manipulate `resp` directly.

What if we were to go back to passing in `resp` and move the logic for setting the body and content type into the serializer itself? In order to avoid breaking existing serializers, we could introspect the `serializer` passed into `set_error_serializer()` and wrap it in a shim if it does not have a `req` arg.